core/controlplane/config/templates/stack-template.json

```diff
@@ -715,6 +737,27 @@
       "Type": "AWS::AutoScaling::LaunchConfiguration"
     },
     {{end}}
+    {{if .Experimental.NodeDrainer.Enabled }}
+    "{{.Controller.LogicalName}}NodeDrainerLHT" : {
+      "Properties" : {
+        "DisplayName" : "SNS topic for the {{.Controller.LogicalName}} node drainer ASG lifecycle hook"
+      },
+      "Type" : "AWS::SNS::Topic"
+    },
+    "{{.Controller.LogicalName}}NodeDrainerLH" : {
+      "Properties" : {
+        "AutoScalingGroupName" : {
+          "Ref": "{{.Controller.LogicalName}}"
+        },
+        "DefaultResult" : "CONTINUE",
+        "HeartbeatTimeout" : "{{.Experimental.NodeDrainer.DrainTimeoutInSeconds}}",
+        "LifecycleTransition" : "autoscaling:EC2_INSTANCE_TERMINATING",
+        "NotificationTargetARN" : { "Ref": "{{.Controller.LogicalName}}NodeDrainerLHT" },
+        "RoleARN" : { "Fn::GetAtt": [ "IAMRoleController", "Arn" ] }
```

Comment:
Topic is indeed dedicated to draining hook, but you give `autoscaling.amazonaws.com` permissions to do much more than publishing to that topic.  Tight explicit permissions is always a safe play.

To reduce size you can create single SNS topic and IAM role to publish there in a root template and pass both to controller and worker templates